lib/render/SliceRenderer.cpp

```diff
@@ -231,11 +231,15 @@ int SliceRenderer::_resetBoxCache() {
         return rc;
     }
     
-    // Moving domain allows area outside of data to be selected
+    // Moving domains do not update Box classes as they change their extents,
+    // so our Box becomes invalid.  We need to store valid extents for efficient
+    // sampling - IE we don't want to sample outside of the domain.
```

Comment:
I'm not sure that this is the right fix. It seems like the cached box class is being modified internally to keep the box inside the domain. To me it seems like this violates the notion of a cache. Secondly, according to the comment in the code it appears that the reason _cacheParams is being modified is to improve efficiency. I don't think performance is an issue here. Bottom line is I think we should consider removing the for loop that changes _cacheParams rather than modifying the user's selected region. The latter may lead to unintended consequences when the domain moves back to whence the region was selected.

Reply:
I see what you mean.  I considered removing the domain/Box check, but chose this fix because I'm concerned about regressions.  

I'm not sure if this Box/Domain validation was a premature optimization, or if it was needed for another reason.  I don't think that this bug (#2157) was an easy one to find, and removing this code will require careful testing of some fringe edge cases.  

I think that setting the Box is less likely to break anything until we better support moving domains.  Just my 2c.

Let me know which you prefer and I'll move forward with your decision.

Reply:
I appreciate concerns over regressions. However, in this case any regressions would be strictly limited to the slice renderer. I say we fix it the "right" way, and test it well

Reply:
Just chiming in since my review was requested, I would be against an "update cache" function modifying the thing it is caching.

Reply:
Thanks guys.  Please re-review when ready.